<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1083</width>
    <height>894</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="gbUnit">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>351</width>
      <height>851</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Unit</string>
    </property>
    <widget class="QComboBox" name="cbPort">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>20</y>
       <width>150</width>
       <height>30</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbPort">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Port</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="btnConnect">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>20</y>
       <width>91</width>
       <height>71</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
    <widget class="QLabel" name="lbBaudRate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>BaudRate</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QComboBox" name="cbBaudRate">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>60</y>
       <width>150</width>
       <height>30</height>
      </rect>
     </property>
    </widget>
    <widget class="QGroupBox" name="gbCurrentPos">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>331</width>
       <height>221</height>
      </rect>
     </property>
     <property name="title">
      <string>Current Position</string>
     </property>
     <widget class="QLabel" name="lbCurrentPosJoint4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Joint4</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbCurrentPosJoint1">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>30</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Joint1</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbCurrentPosJoint2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Joint2</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="tbCurrentPosJoint1">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>30</y>
        <width>220</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="tbCurrentPosJoint4">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>180</y>
        <width>220</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="lbCurrentPosJoint3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Joint3</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="tbCurrentPosJoint3">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>130</y>
        <width>220</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="tbCurrentPosJoint2">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>80</y>
        <width>220</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="gbCurrentPos_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>340</y>
       <width>331</width>
       <height>221</height>
      </rect>
     </property>
     <property name="title">
      <string>Joint Control</string>
     </property>
     <widget class="QLabel" name="lbTargetPosJoint4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Joint4</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbTargetPosJoint1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Joint1</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lbTargetPosJoint2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Joint2</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="tbTargetPosJoint1">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>30</y>
        <width>140</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="tbTargetPosJoint4">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>180</y>
        <width>140</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="lbTargetPosJoint3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Joint3</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="tbTargetPosJoint3">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>130</y>
        <width>140</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="tbTargetPosJoint2">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>80</y>
        <width>140</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btnMoveJoint1">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>20</y>
        <width>101</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Move</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnMoveJoint2">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>70</y>
        <width>101</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Move</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnMoveJoint3">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>120</y>
        <width>101</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Move</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnMoveJoint4">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>170</y>
        <width>101</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Move</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbSequence">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>620</y>
      <width>641</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Sequence Box</string>
    </property>
    <widget class="QPushButton" name="btnSeq1">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>50</y>
       <width>141</width>
       <height>81</height>
      </rect>
     </property>
     <property name="text">
      <string>MovetoTarget</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSeq2">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>50</y>
       <width>141</width>
       <height>81</height>
      </rect>
     </property>
     <property name="text">
      <string>Sequence 2</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSeq2_2">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>50</y>
       <width>141</width>
       <height>81</height>
      </rect>
     </property>
     <property name="text">
      <string>seq3</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbInitParam">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>110</y>
      <width>641</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Inital Position</string>
    </property>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint2_3">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>100</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbInitPos_Z">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Init Z</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint1_3">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>40</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbInitPos_X">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Init X</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint2_4">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>100</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbInitPos_Rx">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>100</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Init Rx</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint1_4">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>40</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbInitPos_Y">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>40</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Init Y</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="btnMoveInitPos">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>30</y>
       <width>80</width>
       <height>110</height>
      </rect>
     </property>
     <property name="text">
      <string>Move</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSaveInitPos">
     <property name="geometry">
      <rect>
       <x>550</x>
       <y>30</y>
       <width>70</width>
       <height>110</height>
      </rect>
     </property>
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbWaitParam">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>280</y>
      <width>641</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Wait Position</string>
    </property>
    <widget class="QLabel" name="lbWaitPos_Z">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Wait Z</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint1_5">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>40</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbWaitPos_X">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Wait X</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="lbWaitPos_Y">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>40</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Wait Y</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint2_5">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>100</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint1_6">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>40</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint2_6">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>100</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbWaitPos_Rx">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>100</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Wait Rx</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="btnMoveWaitPos">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>30</y>
       <width>80</width>
       <height>110</height>
      </rect>
     </property>
     <property name="text">
      <string>Move</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSaveWaitPos">
     <property name="geometry">
      <rect>
       <x>550</x>
       <y>30</y>
       <width>70</width>
       <height>110</height>
      </rect>
     </property>
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbTargetParam">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>450</y>
      <width>641</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Target Position</string>
    </property>
    <widget class="QLabel" name="lbTargetPos_Z">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Target Z</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint1_7">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>40</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbTargetPos_X">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Target X</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="lbTargetPos_Y">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>40</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Target Y</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint2_7">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>100</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint1_8">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>40</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="tbCurrentPosJoint2_8">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>100</y>
       <width>140</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbTargetPos_Rx">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>100</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Target Rx</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="btnMoveTargetPos">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>30</y>
       <width>80</width>
       <height>110</height>
      </rect>
     </property>
     <property name="text">
      <string>Move</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSaveTargetPos">
     <property name="geometry">
      <rect>
       <x>550</x>
       <y>30</y>
       <width>70</width>
       <height>110</height>
      </rect>
     </property>
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btnConnect</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>connect_device()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>288</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnMoveJoint1</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>move_joint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>285</x>
     <y>382</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnMoveJoint2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>move_joint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>444</y>
    </hint>
    <hint type="destinationlabel">
     <x>360</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnMoveJoint3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>move_joint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>270</x>
     <y>487</y>
    </hint>
    <hint type="destinationlabel">
     <x>451</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnMoveJoint4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>move_joint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>545</y>
    </hint>
    <hint type="destinationlabel">
     <x>550</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnMoveInitPos</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>move_joint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>856</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>821</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnMoveWaitPos</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>move_joints()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>856</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>904</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnMoveTargetPos</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>move_joints()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>854</x>
     <y>518</y>
    </hint>
    <hint type="destinationlabel">
     <x>724</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSaveTargetPos</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>964</x>
     <y>546</y>
    </hint>
    <hint type="destinationlabel">
     <x>1031</x>
     <y>441</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>connect_device()</slot>
  <slot>move_joint()</slot>
  <slot>move_joints()</slot>
  <slot>move_L()</slot>
  <slot>move_J()</slot>
  <slot>save()</slot>
 </slots>
</ui>
